File: src/Game.java
import java.io.File;
import java.nio.file.Files;
import java.util.LinkedList;
import java.util.List;
import java.util.stream.Collectors;

public class Game {

    public static List<String> WORDS;

    static {
        try {
            WORDS = Files.readAllLines(new File("./res/words.txt").toPath());
        } catch (Exception e) {
        }
    }

    public static void main(String[] args) {
        Game game = new Game(); System.out.println(game.getAnswer()); System.out.println(game.isValidWord("adieu")); System.out.println(game.isValidWord("rodeo"));
    }

    private final String answer;

    private final List<Guess> guesses = new LinkedList<>();

    public Game() {
        answer = WORDS.get((int) (WORDS.size() * Math.random()));
    }

    public String getAnswer() {
        return answer;
    }

    public void guess(String word) {

        Result[] results = new Result[word.length()];

        List<Character> stack = answer.chars().mapToObj(e -> (char) e).collect(Collectors.toList());

        for (int i = 0; i < word.length(); i++) {
            if (word.charAt(i) == answer.charAt(i)) {
                results[i] = Result.CORRECT;
            } else if (stack.contains(word.charAt(i))) {
                results[i] = Result.PARTIAL; stack.remove((Character) word.charAt(i));
            }
        }

        guesses.add(new Guess(word, results));
    }

    public boolean isValidWord(String word) {

        return guesses.parallelStream().noneMatch(guess -> {

            List<Character> stack = word.chars().mapToObj(e -> (char) e).collect(Collectors.toList());

            for (int i = 0; i < guess.word().length(); i++) {

                switch (guess.results()[i]) {
                    case WRONG -> {
                        if (stack.contains(guess.word().charAt(i))) {
                            return true;
                        }
                    } case PARTIAL -> {
                        if (guess.word().charAt(i) == word.charAt(i)) {
                            return true;
                        } if (!stack.contains(guess.word().charAt(i))) {
                            return true;
                        } stack.remove((Character) guess.word().charAt(i));
                    } case CORRECT -> {
                        if (word.charAt(i) != guess.word().charAt(i)) {
                            return true;
                        }
                    }
                }
            }

            return false;
        });
    }
}

File: game.py
import json
import random
from turtle import clone
# game emulatopr

def load_words() -> list[str]:
    with open('./res/words.json') as file:
        return json.load(file)

WORD_LIST = load_words()

DEFAULT = 0
WRONG = 1
PARTIAL = 2
CORRECT = 3

class Game:
    def __init__(self) -> None:
        self.guesses = []
        self.word = random.choice(WORD_LIST)
        self.valid_words = WORD_LIST.copy()
        pass

    def guess(self,word: str):
        if len(word) != len(self.word) or word not in WORD_LIST:
            return None

        stack = [c for c in self.word]
        values = [WRONG] * len(word)
        for i in range(len(self.word)):
            if word[i] == self.word[i]:
                values[i] = CORRECT
                stack.remove(word[i])
        for i in range(len(self.word)):
            if word[i] in stack:
                values[i] = PARTIAL
                stack.remove(word[i])

        self.guesses.append(word)

        # filter valid words
        filtered = []
        for vword in self.valid_words:
            filter = True
            stack = [c for c in vword]
            for i in range(len(vword)):
                # if vword[i] == word[i] and values[i] != CORRECT:
                #     filter = False
                # HELP
                #     break
                if values[i] == WRONG and word[i] in vword:
                    filter = False
                    break
                if vword[i] != word[i] ^ values[i] == CORRECT:
                    filter = False
                    break
                else:
                    stack.remove(vword[i])
                if values[i] == PARTIAL:
                    if word[i] in stack:
                        stack.remove(word[i])
                    else:
                        filter = False
                        break

            if filter:
                filtered.append(vword)




        self.valid_words = filtered

        return values
